src/socket.ts

```diff
@@ -1,15 +1,15 @@
 import { client as WebSocketClient } from 'websocket';
+import { insertTrade } from './data-storage';
+import { CreateTradeInput } from './types';
 
-export const connectCoinbase = () => {
+export const connectCoinbase = (productIds: string[], channels: string[]) => {
   const cbSocket = new WebSocketClient();
-  const subscribeMessage = JSON.stringify({
-    type: "subscribe",
-    channels: [
-      {
-        name: "ticker",
-        product_ids: ["ETH-USD"],
-      },
-    ],
+  const subscribeMessage = { type: "subscribe", channels: new Array() };
+  channels.forEach(c => {

```

Comment:
Flavor: It is preferential to use full names when assigning variables, even for something this simple, so that the code itself is self documenting. The amount of characters saved does not make up for the amount of information lost.